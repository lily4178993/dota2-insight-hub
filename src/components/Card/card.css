.card-container {
  position: relative;
  width: 100%;
  height: 190px;
  padding: 0.5em;
}

.card-image {
  position: absolute;
  top: 20%;
  left: 50%;
  transform: translate(-50%, -20%);
  width: 140px;
  height: 140px;
}

.card-image img {
  position: relative;
  width: 100%;
  height: 100%;
  object-fit: cover;
  filter: brightness(10%) opacity(80%);
  transition: all 1s cubic-bezier(0.16, 1, 0.3, 1);
}

.card-container:hover .card-image img,
.card-container:focus-within .card-image img {
  filter: brightness(100%) opacity(100%);
  transform: scale(1.1);
}

.card-container.odd {
  background: var(--color-blue-3);
}

.card-container.even {
  background: var(--color-blue-2);
}

.card-container button {
  position: absolute;
  top: 0.5em;
  right: 1em;
}

.card-container button i {
  font-size: 1.4em;
}

.card-content {
  position: absolute;
  bottom: 1.4em;
  right: 1em;
  text-align: right;
}

@media (max-width: 768px) {
  .card-container {
    width: 100%;
  }

  .card-container:nth-child(4n - 3),
  .card-container:nth-child(4n - 4) {
    background: var(--color-blue-2);
  }

  .card-container:nth-child(4n - 2),
  .card-container:nth-child(4n - 1) {
    background: var(--color-blue-3);
  }
}
