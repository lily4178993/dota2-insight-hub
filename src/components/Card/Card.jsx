import React from 'react';
import PropTypes from 'prop-types';
import './card.css';

function Card({
  cardImage, cardTitle, cardCount,
}) {
  return (
    <div className="card-container">
      <button type="button" aria-label="More details" title="More details">
        <i className="bx bx-right-arrow-circle" />
      </button>
      <div className="card-image">
        <img src={cardImage} alt={typeof cardTitle === 'function' ? 'N/A' : cardTitle} />
      </div>
      <div className="card-content">
        <h2>{typeof cardTitle === 'function' ? 'N/A' : cardTitle}</h2>
        <span>{cardCount}</span>
      </div>
    </div>
  );
}

Card.defaultProps = {
  cardImage: '',
  cardTitle: '',
};

Card.propTypes = {
  cardCount: PropTypes.oneOfType([
    PropTypes.number,
    PropTypes.string,
  ]).isRequired,
  cardImage: PropTypes.string,
  cardTitle: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.func,
  ]),
};

export default Card;
