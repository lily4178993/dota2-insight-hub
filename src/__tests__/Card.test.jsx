import React from 'react';
import { render } from '@testing-library/react';
import Card from '../components/Card/Card';

test('Card component renders correctly', () => {
  const props = {
    cardImage: 'image.jpg',
    cardTitle: 'Sample Card',
    cardCount: 42,
  };

  // eslint-disable-next-line react/jsx-props-no-spreading
  const { asFragment } = render(<Card {...props} />);
  expect(asFragment()).toMatchSnapshot();
});

test('Card component renders with parser image correctly', () => {
  const props = {
    cardImage: 'image.jpg',
    cardTitle: 'Sample Card',
    cardCount: 42,
    isParser: true,
  };

  // eslint-disable-next-line react/jsx-props-no-spreading
  const { asFragment } = render(<Card {...props} />);
  expect(asFragment()).toMatchSnapshot();
});
