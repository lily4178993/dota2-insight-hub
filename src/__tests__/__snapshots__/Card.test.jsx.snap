// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="card-container"
  >
    <button
      title="More details"
      type="button"
    >
      <i
        class="bx bx-right-arrow-circle"
      />
    </button>
    <div
      class="card-image"
    >
      <img
        alt="Sample Card"
        src="image.jpg"
      />
    </div>
    <div
      class="card-content"
    >
      <h2>
        Sample Card
      </h2>
      <span>
        42
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Card component renders with parser image correctly 1`] = `
<DocumentFragment>
  <div
    class="card-container"
  >
    <button
      title="More details"
      type="button"
    >
      <i
        class="bx bx-right-arrow-circle"
      />
    </button>
    <div
      class="card-image"
    >
      <img
        alt="Sample Card"
        src="https://cdn.cloudflare.steamstatic.com/apps/dota2/videos/dota_react/heroes/renders/image.jpg.png"
      />
    </div>
    <div
      class="card-content"
    >
      <h2>
        Sample Card
      </h2>
      <span>
        42
      </span>
    </div>
  </div>
</DocumentFragment>
`;
